src/abstractdataframe/show.jl

```diff
@@ -60,11 +60,25 @@ end
 #'
 #' ourshowcompact(stdout, "abc")
 #' ourshowcompact(stdout, 10000)
-ourshowcompact(io::IO, x::Any) = show(IOContext(io, :compact=>true), x) # -> Void
+ourshowcompact(io::IO, x::Any) = show(IOContext(io, :compact=>true, :typeinfo=>typeof(x)),
+                                      x) # -> Void
 ourshowcompact(io::IO, x::AbstractString) = escape_string(io, x, "") # -> Void
 ourshowcompact(io::IO, x::Symbol) = ourshowcompact(io, string(x)) # -> Void
 ourshowcompact(io::IO, x::Nothing) = nothing 
 
+#' @description
+#'
+#' Returns compact string representation of type T
+function compacttype(T::Type, maxwidth::Int)
+    maxwidth = max(maxwidth, 2)
+    T === Any && return "Any"
+    T === Missing && return "Missing"
+    T >: Missing && return compacttype(Base.nonmissingtype(T), maxwidth-1) * "⍰"
+    sT = string(T)
+    length(sT) ≤ maxwidth && return sT
+    match(Regex("^.\\w{0,$(maxwidth-2)}"), string(T)).match * "…"
```

Comment:
Could use `sT`.

Reply:
Fixed. Actually this was why I have introduced `sT`, but then forgotten to use it.